python/pyspark/sql/streaming.py

```diff
@@ -186,7 +186,7 @@ def exception(self):
             je = self._jsq.exception().get()
             msg = je.toString().split(': ', 1)[1]  # Drop the Java StreamingQueryException type info
             stackTrace = '\n\t at '.join(map(lambda x: x.toString(), je.getStackTrace()))
-            return StreamingQueryException(msg, stackTrace)
+            return StreamingQueryException(msg, stackTrace, je.getCause())
```

Comment:
To be super safe, should you check that `je.getCause()` exists?

Reply:
Cause check (not None) is inside `class CapturedException`. Do you suggest something different here?

Reply:
I mean if `je.getCause()` is `None`, shouldn't you be using `je` so that the error is not lost?